modules/react-components/src/components/renderer/markdown/components/markdown-link.tsx

```diff
@@ -0,0 +1,171 @@
+    PrimaryButton
+} from "@wso2is/react-components";
+import { saveAs } from "file-saver";
+import React, { FunctionComponent, ReactElement, useMemo } from "react";
+import useGlobalMarkdown from "../../../../hooks/use-global-markdown";
+
+const DEFAULT_DOWNLOAD_FILE_NAME: string = "download";
+
+/**
+ * Component types for rendering the link.
+ */
+enum ComponentTypes {
+    LINK = "link",
+    BUTTON = "button"
+}
+
+/**
+ * Props interface for the `MarkdownLink` component.
+ */
+interface MarkdownLinkProps extends MarkdownCustomComponentPropsInterface<"a"> {
+    /**
+     * Custom attributes supplied by the 'rehype-attr' plugin.
+     */
+    "data-config"?: {
+        /**
+         * Flag to determine whether the link is an external link.
+         */
+        external?: boolean;
+        /**
+         * Flag to determine whether the link is a download link.
+         */
+        download?: boolean;
+        /**
+         * If the link is a download link, specify the name for the downloaded file.
+         */
+        fileName?: string;
+        /**
+         * Download the string content as a file.
+         * This will take precedence over href.
+         * Content should have a Base64 encoded string.
+         */
+        content?: string;
+        /**
+         * The type that should be assigned to the above content file.
+         */
+        type?: string;
+        /**
+         * Component type of the link element.
+         */
+        as?: ComponentTypes;
+    };
+}
+
+/**
+ * Markdown custom component for the link element.
+ *
+ * @param Props - Props to be injected into the component.
+ */
+const MarkdownLink: FunctionComponent<MarkdownLinkProps> = ({
+    href,
+    title,
+    children,
+    "data-config": dataConfig,
+    "data-componentid": componentId = "custom-markdown-link"
+}: MarkdownLinkProps): ReactElement => {
+
+    const { onHandleInternalUrl } = useGlobalMarkdown();
+
+    /**
+     * Initiate the download process.
+     */
+    const initDownload = async (): Promise<void> => {
+        if (dataConfig?.content && dataConfig?.type) {
+            const blob: Blob = new Blob([ atob(dataConfig?.content) ], {
+                type: dataConfig?.type
+            });
+
+            if (dataConfig?.fileName) {
+                saveAs(blob, dataConfig?.fileName);
+            } else {
+                saveAs(blob, DEFAULT_DOWNLOAD_FILE_NAME);
+            }
+
+            return;
+        }
+
+        const response = await fetch(href);
+
+        // Check if the response is ok (status is in the range 200-299).
+        if (response.ok) {
+            // Convert the response to a Blob.
+            const blob = await response.blob();
+
+            if (blob) {
+                if (dataConfig?.fileName) {
+                    saveAs(blob, dataConfig?.fileName);
+                } else {
+                    saveAs(blob, DEFAULT_DOWNLOAD_FILE_NAME);
+                }
+            }
```

Comment:
add a conols.error for errors

Reply:
Currently, we are not allowed to use `console.error` in our code. We have an ESLint rule that enforces this.